### PR TITLE
Added ports to the scatter and enable usage of variables into the scatter

### PR DESCRIPTION
**Pull request by daniilsavchuk**
## General idea

Since [PR#20](https://github.com/epam/pipeline-builder/pull/20) was been merged it become possible to have a connections and ports in the scatter (if/while). This possibility make us able to propose solution for the following issues: #9 , #11 , #15 , and rewrite temporary fix for #16 , #21 . Currently pipeline-builder does not support the inputs in scatter(if/while) and does not show the scatter iteration item as a port. This pull request implements this features and hopefully fixes some bugs ( #15 , #16 , #21  ).

**In this PR it will look like:**
GroupInputs

## Changes

- Removed variable usege limitation for scatter 
- Enabled ports for scatter, if, while
- Enabled connections between scatter/if/while with other things
- Updated WDL generator code
- Updated tests due to the new scatter/if/while functionality.

## Scripts affected
- [ExtractSamHeadersWf_170107](https://raw.githubusercontent.com/broadinstitute/wdl/develop/scripts/broad_dsde_workflows/ExtractSamHeadersWf_170107.wdl)
- [RevertRGBamsToPairedFastQsWf_170107](https://raw.githubusercontent.com/broadinstitute/wdl/develop/scripts/broad_dsde_workflows/RevertRGBamsToPairedFastQsWf_170107.wdl)
- [ValidateBamsWf_170107](https://raw.githubusercontent.com/broadinstitute/wdl/develop/scripts/broad_dsde_workflows/ValidateBamsWf_170107.wdl)
- [PublicPairedSingleSampleWf_170412](https://raw.githubusercontent.com/broadinstitute/wdl/develop/scripts/broad_pipelines/PublicPairedSingleSampleWf_170412.wdl)

### CODE REVIEW

File: src/model/Group.js [hunk 1]
```diff
@@ -27,9 +27,7 @@ class Group extends Step {
    * @param {object} [config={}] - Group configuration containing group meta information.
    */
   constructor(name, type, config = {}) {
-    if (!(config instanceof Action)) {
-      config.canHavePorts = false;
-    } else {
+    if (config instanceof Action) {
```
Comment (voidest) [inline src/model/Group.js:30]: Do we still unable to create groups from actions?

File: test/model/GroupTest.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ describe('model/Group', () => {
     });
 
     it('throws error when try to create with ports', () => {
```
Comment (voidest) [inline test/model/GroupTest.js]: Should we rename/remove this test?

**Review by voidest — APPROVED** [review]
Seems fine. I just have a couple of questions here.

File: src/model/Group.js (hunk 1 above)
Comment (daniilsavchuk) [reply src/model/Group.js:30]: @voidest currently it looks like there are no sense to create groups from actions. Let me leave it for a while.

File: test/model/GroupTest.js (hunk 2 above)
Comment (daniilsavchuk) [reply test/model/GroupTest.js]: Thank you for the notisance! 

I will correct the name of this test

**Review by sidoruka — APPROVED** [review]

**Pull request merged**